test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -430,13 +430,18 @@ protected ScriptService getMockScriptService() {
         return null;
     }
 
+    /**
+     * Should we aggregate on each leaf individually and reduce all results together,
+     * or aggregate across the whole index and reduce once on the single reseulting agg
+     */
+    protected boolean splitLeavesIntoSeparateAggregators() {
+        return randomBoolean();
+    }
```

Comment:
@not-napoleon and I were just talking about this sort of thing. It makes sense here but I wonder if @not-napoleon has a different idea?

Reply:
Yeah, my other idea is "let's actually finish cleaning up aggregations tests."  But that's weeks of work with very little gain, and there's no sense in blocking this PR on that work.